# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Função f(x) = x³: 
f = lambda x: np.power(x, 3)

def calcula_area(a, b, func, frame_num):
    num_bins=frame_num+1 # Quantidade de retângulos
    delta = (b-a)/num_bins # largura do retângulo
    x_cols = np.linspace(a, b-delta, num_bins)+delta/2 # Divisão do eixo X
    y_cols = f(x_cols) # Alturas dos retângulos
    return sum(delta*y_cols)

In [5]:
calcula_area(1, 2, f, 100) #Usando 100 retângulos

3.7499632388981476

De forma analítica:

$$f(x) = F'(x)$$
$$f(x) = x^3$$
$$F(x) = \int x^3 dx = \frac{x^4}{4}$$

Entre 1 e 2:

$$F(2)-F(1) = \frac{2^4}{4}-\frac{1^4}{4} = \frac{16}{4} - \frac{1}{4} = \frac{15}{4} = 3.75$$

O valor calculado numericamente se aproxima do valor teórico mas apenas será o mesmo caso sejam utilizados infinitos retângulos para o cálculo da área. 
